arch/risc-v/src/common/Toolchain.defs

```diff
@@ -210,7 +210,9 @@ ifeq ($(CONFIG_RISCV_TOOLCHAIN),GNU_RVG)
   endif
 
   ifeq ($(CONFIG_ARCH_RV_ISA_ZICSR_ZIFENCEI),y)
-    ARCHCPUEXTFLAGS := $(ARCHCPUEXTFLAGS)_zicsr_zifencei
+    ifeq ($(shell expr $(GCCVER) \>= 12), 1)
```

Comment:
So we do not check for `CONFIG_ARCH_TOOLCHAIN_CLANG` here?

Reply:
It seems that CONFIG_ARCH_TOOLCHAIN_CLANG needs to be added. the latest changes have already added the judgment. please review again，Thanks.